libsolidity/analysis/FunctionCallGraph.cpp

```diff
@@ -0,0 +1,109 @@
+#include <libsolidity/analysis/FunctionCallGraph.h>
+//#include <libsolutil/CommonData.h>
+
+#include <boost/range/adaptor/reversed.hpp>
+
+using namespace solidity::frontend;
+
+void FunctionCallGraphBuilder::analyze()
+{
+	for (auto const* contract: m_currentContract->annotation().linearizedBaseContracts  | boost::adaptors::reversed)
+		if (contract->constructor())
+			contract->constructor()->accept(*this);
+
+	for (VariableDeclaration const* stateVar: m_currentContract->stateVariablesIncludingInherited())
+		if (stateVar->value())
+		{
+			m_currentNode = stateVar;
+			stateVar->value()->accept(*this);
+			m_mapping.insert(std::pair(stateVar, std::move(m_currentMapping)));
+		}
+}
+
+
+bool FunctionCallGraphBuilder::visit(FunctionDefinition const& _functionDefinition)
+{
+	return analyseCallable(_functionDefinition);
+}
+
+
+bool FunctionCallGraphBuilder::visit(ModifierDefinition const& _modifierDefinition)
+{
+	return analyseCallable(_modifierDefinition);
+}
+
+
+bool FunctionCallGraphBuilder::visit(Identifier const& _identifier)
+{
+	if (auto const callableDef = dynamic_cast<CallableDeclaration const*>(_identifier.annotation().referencedDeclaration))
+		if (m_currentMapping.insert(callableDef).second)
+			findFinalOverride(callableDef)->accept(*this);
```

Comment:
Can `findFinalOverride` return `nullptr`?

Reply:
only for declarations that don't exist in the contract which in this case is not possible